Build.sh improvement

jonisc opened the pull request:
Now you just have to execute ./build.sh versionfolder to create a new .deb. Basically just an improvement. 
Tried fixing .DS_Store issues (delete all .DS_Store files in Gandalf folder in build.sh) but I need some confirmation by @mehulrao. 

Several small improvements. Not only @1Conan 's improvements. 
Will check if folder exists and files are present. You can argue about this, but it's just better to do so in my opinion.

1Conan commented on Gandalf/build.sh:
```diff
@@ -84,9 +129,13 @@ echo "Copying the DEBIAN scripts"
 
-echo "If you have anything else you need to put into Gandalf, now's the time. You have 15 seconds from the moment this message appears."
-sleep 15
+find . -name '.DS_Store' -type f -delete
```
Nothing different with my `find`. You just changed `-exec rm {}`. `-exec rm {}` is actually much more recommended.

jonisc replied:
Ok. Then I'll use your find.

jonisc replied:
Question you use `find . -type f -name '.DS_STORE' -exec rm {} +`
What's the + for?

1Conan commented on Gandalf/build.sh:
```diff
@@ -1,36 +1,81 @@
+RED=$(tput setaf 1)
+
+# Check if started without parameters
+if [ "${1}" = "" ]; then
```
https://google.github.io/styleguide/shell.xml

`${1}` should be `$1`.

Also, I believe you're declaring a variable.

jonisc commented:
@1Conan ok. So I fixed many things. But I can't put all the constants & variables to the top: Otherwise there'll be some cat errors (cat can't find file or folder -> error). If I abort the script before defining e.g. PKG_CONFLICTS there won't be this errors. It's just cosmetic… but if you have a better idea you can improve the branch.

jonisc commented:
Improvements by @1Conan in #51.

jonisc commented:
Ok. He wants to merge into builds branch and not into master.

jonisc commented:
@mehulrao: There´s something wrong with your App: If you compile Gandalf (102/x) via eg. Ubuntu there´ll be errors: there are then 2 Folders. One called Gandalf.app with capital G, and an other called gandalf.app with a lowercase g. So Ubuntu copies the Gandalf.app and if you install it on iOS the app won´t show. I´ll compile your app via my mac, and we´ll see what´s happening...

jonisc commented:
Didn't compile your app. Just extracted it from a the latest official .deb. 

There were some errors. Hopefully I squashed them now. One error I think we must discuss is this one:
[Dpkg wants to remove /var/mobile/Downloads](https://imgur.com/a/uhUqd)

EthanRDoesMC commented:
That's a dpkg thing that dpkg does. If downloads were empty, it'd try to remove /mobile. Which it can't. Because there are things in it.

EthanRDoesMC commented:
To build Gandalf, you need macOS or an OS that recognizes the .app extension in the same way. But since macOS is not preloaded with the stuff required to make it, there will need to be YET ANOTHER FREAKING SCRIPT (lol) to install the stuff on macOS.

EthanRDoesMC commented:
I figure this is okay for tonight.

jonisc commented:
I could build it on Ubuntu ;-). You just have to make sure that there's only a Gandalf.app folder.

jonisc commented:
@ Everyone @EthanRDoesMC @mehulrao
Just to clarify for anybody. 
Attention: syntax has changed: now you must cd to the folder where build.sh is and execute it with ./build.sh version:

`./build.sh x` will build GandalfX, `./build.sh 102` will build Gandalf102.
